service/controller/clusterapi/v21/resource/updatemachinedeployments/create.go

```diff
@@ -100,7 +50,7 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		// Syncing the Provider Operator version. For instance aws-operator,
 		// kvm-operator or the like.
 		{
-			l := versionLabel
+			l := fmt.Sprintf("%s-operator.giantswarm.io/version", r.provider)
```

Comment:
It is way simpler to configure the provider at runtime and construct the version label using it than looking up nodes from the Tenant Cluster API. We just don't need the remote dependency anymore for keeping the resource implementation functioning properly.